src/components/Tree/Tree.cy.tsx

```diff
@@ -34,37 +66,244 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTree={onDropStub} />);
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID)
+            .first()
+            .focus()
+            .realPress('Space')
+            .realPress('ArrowDown')
+            .realPress('ArrowRight')
+            .realPress('Space');
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('calls onDrop on the root level when controlled by the TreeItem', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).first().realMouseDown().realMouseMove(0, 40).realMouseUp();
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('calls onDrop on level 1 when controlled by the TreeItem', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(4).realMouseDown().realMouseMove(0, -40).realMouseUp();
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('calls onDrop when moving to an inside item controlled by the TreeItem', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(6).realMouseDown().realMouseMove(40, -40).realMouseUp();
+        cy.get('@onDropStub').should('not.have.been.called');
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(6).realMouseDown().realMouseMove(40, -80).realMouseUp();
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('calls onDrop when moving to an outside item controlled by the TreeItem', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(5).realMouseDown().realMouseMove(-40, 80).realMouseUp();
+        cy.get('@onDropStub').should('not.have.been.called');
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(5).realMouseDown().realMouseMove(-40, 40).realMouseUp();
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('allows adding an item dynamically to the Tree', () => {
+        const newNode: TreeItemMock = {
+            id: '4',
+            label: 'New Added Item',
+        };
+        const rootItemsLength = treeItemsMock.length;
+
+        cy.mount(<TreeComponent />).then(({ rerender }) => {
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength);
+            treeItemsMock.push(newNode);
```

Comment:
aren't you affecting other test with this mutation of treeItemsMock?

Reply:
The pushed item gets popped in the next test so it should be fine, no?